src/core/MeshBVH.js

```diff
@@ -943,13 +1084,74 @@ export class MeshBVH {
 
 		);
 
-		return Math.sqrt( closestDistanceSq );
+		const closestDistance = Math.sqrt( closestDistanceSq );
 
-	}
+		const indices = this.geometry.getIndex().array;
+		const positions = this.geometry.getAttribute( 'position' );
+		const uvs = this.geometry.getAttribute( 'uv' );
+
+		const a = indices[ closestDistanceTriIndex * 3 ];
+		const b = indices[ closestDistanceTriIndex * 3 + 1 ];
+		const c = indices[ closestDistanceTriIndex * 3 + 2 ];
 
-	distanceToPoint( point, minThreshold, maxThreshold ) {
+		tempV1.fromBufferAttribute( positions, a );
+		tempV2.fromBufferAttribute( positions, b );
+		tempV3.fromBufferAttribute( positions, c );
 
-		return this.closestPointToPoint( point, null, minThreshold, maxThreshold );
+		let uv = null;
+		if ( uvs ) {
+
+			tempUV1.fromBufferAttribute( uvs, a );
+			tempUV2.fromBufferAttribute( uvs, b );
+			tempUV3.fromBufferAttribute( uvs, c );
+
+			if ( target && target.uv ) uv = target.uv;
+			else uv = new Vector2();
+			Triangle.getUV( temp1, tempV1, tempV2, tempV3, tempUV1, tempUV2, tempUV3, uv );
+
+		}
+
+		if ( target ) {
+
+			if ( target.isVector3 ) {
+
+				target.copy( temp1 );
+				return closestDistance;
+
+			}
+
+			if ( ! target.point ) target.point = new Vector3();
+			target.point.copy( temp1 );
+			target.distance = closestDistance;
+			if ( ! target.face ) target.face = { };
+			target.face.a = a;
+			target.face.b = b;
+			target.face.c = c;
+			target.face.materialIndex = 0;
+			if ( ! target.face.normal ) target.face.normal = new Vector3();
+			Triangle.getNormal( tempV1, tempV2, tempV3, target.face.normal );
+			if ( ! target.uv ) target.uv = new Vector2();
+			target.uv.copy( uv );
+
+			return target;
```

Comment:
A lot of this looks like it could be reused between the two cases in `closestPointToGeometry` and this function -- should we make a utility function for constructing this based on a point, geometry, and triangle index? It could go in the [utils/TriangleUtilities.js file](https://github.com/gkjohnson/three-mesh-bvh/blob/master/src/utils/TriangleUtilities.js). The function could modify an existing object if it's passed in or return a new one if there's no object provided.